### PR TITLE
Add cache to mobile_config client with default ttl of 60 minutes

### PR DESCRIPTION
**Pull request by bbalser**

### CODE REVIEW

File: mobile_config/src/client/mod.rs [hunk 1]
```diff
@@ -48,23 +68,32 @@ impl gateway_info::GatewayInfoResolver for Client {
...
-                response.verify(&self.config_pubkey)?;
-                response.info.map(gateway_info::GatewayInfo::from)
-            }
-            Err(status) if status.code() == tonic::Code::NotFound => None,
-            Err(status) => Err(status)?,
-        };
-        Ok(response)
+        if let Some(cached_response) = self.cache.get(address).await {
+            Ok(cached_response.value().clone())
+        } else {
```
Comment (maplant) [inline mobile_config/src/client/mod.rs:73]: I think this is one place where an early return would be preferred

Comment (bbalser) [reply mobile_config/src/client/mod.rs]: Agreed cb96028

**Review by maplant — APPROVED** [review]

File: mobile_config/src/client/mod.rs [hunk 2]
```diff
@@ -21,22 +24,39 @@ pub enum ClientError {
...
 }
 
-#[derive(Clone, Debug)]
+#[derive(Clone)]
 pub struct Client {
     pub client: mobile_config::GatewayClient<Channel>,
     signing_key: Arc<Keypair>,
     config_pubkey: PublicKey,
     batch_size: u32,
+    cache: Arc<Cache<PublicKeyBinary, Option<gateway_info::GatewayInfo>>>,
```
Comment (jeffgrunewald) [inline mobile_config/src/client/mod.rs:33]: the cache doesn't implement Debug i take it...

**Review by jeffgrunewald — APPROVED** [review]

File: mobile_config/src/client/mod.rs (hunk 2 above)
Comment (bbalser) [reply mobile_config/src/client/mod.rs:33]: It does not

**Pull request merged**